### PR TITLE
fix: ES modules for main process

### LINKED ISSUE #403: Deepmerge import is broken in vite/rollup
- [x] Review the documentation: https://docs.sentry.io/platforms/javascript/electron/
- [x] Search for existing issues: https://github.com/getsentry/sentry-electron/issues
- [x] Use the latest release: https://github.com/getsentry/sentry-electron/releases
- [ ] Provide a link to the affected event from your Sentry account

## Versions + Platform

- [x] SDK version - `@sentry/electron@v3.0.0-beta-3`
- [x] Electron version - `electron@v16.0.5`
- [x] Platform - `Windows/macOS/Linux`
- [x] Bundler - `Vite/Rollup`

## Description

Errors:
```bash
Cannot call a namespace ('deepMerge')
```

```bash
Sentry Logger [Error]: SentryError2: Event processing pipeline threw an error, original event will not be sent. Details have been sent as a new event.
Reason: TypeError: deepMerge is not a function
```

The way deepmerge is imported at https://github.com/getsentry/sentry-electron/blob/12058e46133f762bb7bdc873982727a3da813e88/src/common/merge.ts#L3 works incorrectly in Vite/Rollup - https://github.com/TehShrike/deepmerge/issues/102

Similar issue: https://github.com/vitejs/vite/issues/2261#issuecomment-808961321

Fix is to change import to intended: https://github.com/TehShrike/deepmerge/issues/102#issuecomment-384975231
```js
import deepmerge from 'deepmerge'
```

### PR DESCRIPTION
**Pull request by timfish**
- Fixes `deepmerge` import to support ES modules
- Adds an example app built with Vite to ensure no regression
- Fixes `electron-forge` examples due to colors.js "issues"

Fixes #403

### CODE REVIEW

**Comment by timfish** [comment]
The current test failures are due to:
https://github.com/electron-userland/electron-forge/issues/2665

**Comment by artplan1** [comment]
hi @timfish, you can see the issue with `deepmerge` I mentioned in https://github.com/getsentry/sentry-electron/issues/403 when you run `yarn build` in this new example:

```bash
> yarn build

yarn run v1.22.15
warning package.json: No license field
$ yarn run build:main && yarn run build:renderer
warning package.json: No license field
$ vite build --config vite.config.main.js
vite v2.7.10 building for production...
✓ 1 modules transformed.
dist/main/index.js   0.51 KiB
warning package.json: No license field
$ vite build --config vite.config.renderer.js
vite v2.7.10 building for production...
transforming (70) node_modules/@sentry/browser/esm/transports/fetch.jsCannot call a namespace ('deepMerge')
✓ 86 modules transformed.
dist/renderer/index.html           0.33 KiB
dist/renderer/index.40735aa1.js    0.82 KiB
dist/renderer/index.40735aa1.js.map 2.55 KiB
dist/renderer/vendor.d9797905.js   79.10 KiB
dist/renderer/vendor.d9797905.js.map 384.80 KiB
✨  Done in 3.03s.
```

**Comment by timfish** [comment]
That output from `yarn build` looks fine. What am I looking for?

**Comment by artplan1** [comment]
@timfish ok, looks like it doesn't work for me because I use ES6 imports in main

`import { init } from '@sentry/electron/main'`

Rollup build doesn't fail, it throws same warning about deepmerge, but errors aren't sent to sentry.

**Comment by timfish** [comment]
Ah ok, thanks for your help narrowing it down. I'll change this example to use ES imports in main and get it fixed!

File: src/common/merge.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
-/* eslint-disable @typescript-eslint/no-unsafe-member-access */
 import { Event } from '@sentry/types';
-import * as deepMerge from 'deepmerge';
+const deepMerge = require('deepmerge');
```
Comment (AbhiPrasad) [inline src/common/merge.ts:3]: Can we add a comment why we need to have require here?

**Review by AbhiPrasad — APPROVED** [review]

**Pull request merged**